### DISCUSSION TITLE
算法能否支持字符串id

### DISCUSSION THREAD

**Discussion by inurfuture** [alibaba/GraphScope#1185] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/42594720/146484435-350a1458-887e-4702-b5eb-c16e51983681.png)
在载图的文档中看到，id是支持为字符串的，只是会影响存储和算法性能

![image](https://user-images.githubusercontent.com/42594720/146484328-ec3510ac-9c93-426c-8869-cf3854c821ba.png)
根据api文档中描述，例如sssp算法的入参src类型为int

请问sssp算法是否能支持字符串id呢，对性能的影响会有多大呢

**Comment by welcome** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! And a maintainer will get back to you shortly!

**Comment by sighingnow** [comment, marked as answer]
> 请问sssp算法是否能支持字符串id呢

`src` definitely could be a `str` if the oid type of the graph is string.

> 对性能的影响会有多大呢

For built-in app `sssp`, the impact on performance should be ignoreable.

**Reply by inurfuture** [reply]
测试了一下sssp，结果只有dist，如何能获取具体路径呢？
![image](https://user-images.githubusercontent.com/42594720/146515579-2e0c23ba-4dd1-4a0a-8494-fac8c8389718.png)

**Reply by acezen** [reply]
> 测试了一下sssp，结果只有dist，如何能获取具体路径呢？ ![image](https://user-images.githubusercontent.com/42594720/146515579-2e0c23ba-4dd1-4a0a-8494-fac8c8389718.png)

we didn't have a builtin app to get the shortest path now, but you can transfer the graph to graphscope.nx graph, and use `nx.shortest_path` to get the shortest path:
``` python
import graphscope.nx as nx
G = nx.DiGraph(g)  # g is the gs.g you created
nx.shortest_path(G, source=0)
```

**Comment by sighingnow** [comment]
And yes, we should update the document to reflect the truth that here `str` is ok as well.
